packages/sdk/src/oracle/utils.ts

```diff
@@ -82,10 +83,32 @@ export function getFlags(state: State) {
     // ignore
   }
 
-  // TODO: add logic for these
-  flags[Flag.ProposalInProgress] = false;
-  flags[Flag.DisputeInProgress] = false;
-  flags[Flag.ApprovalInProgress] = false;
+  try {
+    flags[Flag.ProposalInProgress] = false;
+    flags[Flag.DisputeInProgress] = false;
+    flags[Flag.ApprovalInProgress] = false;
+    flags[Flag.ChainChangeInProgress] = false;
+
+    // get all active commands
+    const commands = read.filterCommands({ done: false, user: read.userAddress() });
```

Comment:
queries alll commands (contexts) for a user, and checks the type to see if any are in progress. sets flags accordingly